lib/house.rb

```diff
@@ -0,0 +1,62 @@
+class House
+attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    rooms << room
+  end
+
+  def above_market_average?
+    if @price[1..-1].to_i > 500000
+      true
+    else
+      false
+    end
+  end
+
+  def rooms_from_category(category)
+    rooms_category = []
+    @rooms.each do |room|
+      if room.category == category
+        rooms_category << room
+      end
+    end
+    rooms_category
+  end
+
+  def area
+    all_room_areas = []
+    @rooms.each do |room|
```

Comment:
Fantastic job using a combination of each and sum to get the job done in this method. `sum` can take a block to make this even cleaner: 

`@rooms.sum {|room| room.area}`